csrc/bfs.h

```diff
@@ -620,6 +620,64 @@ class BFSWithPermissiveDependence
   }
 };
 
+// Unlike the default BFS behavior, Val is considered ready to
```

Comment:
Ended up not using this, but maybe useful in the future

Reply:
I am fine with keeping it. But isn't `BFS` requires all definitions, but `BFSWithPermissiveDependence` only requires one?

Reply:
`BFS` requires only one of definitions (or uses) to be visited.

https://github.com/NVIDIA/Fuser/blob/main/csrc/bfs.h#L432-L435
https://github.com/NVIDIA/Fuser/blob/main/csrc/bfs.h#L443-L446

Reply:
Oh, I see. So `BFS` requires all inputs/outputs of exprs to be visited, but `BFSWithPermissiveDependence` requires  only one input/output of expr to be visited. So we have:

|                      | All inputs/outputs      | One input/output            |
|----------------------|-------------------------|-----------------------------|
| All uses/definitions | BFSWithStrictDependence |                             |
| One use/definition   | BFS                     | BFSWithPermissiveDependence |

Reply:
That is right. I started prototyping `BFSWithStrictDependence` for this PR but ended up not using it. Not sure if it would be actually useful.